pkg/app/api/api/piped_api.go

```diff
@@ -60,6 +65,9 @@ func NewPipedAPI(ds datastore.DataStore, sls stagelogstore.Store, alss applicati
 		stageLogStore:             sls,
 		applicationLiveStateStore: alss,
 		commandStore:              cs,
+		appPipedCache:             memorycache.NewTTLCache(ctx, 24*time.Hour, 3*time.Hour),
+		deploymentPipedCache:      memorycache.NewTTLCache(ctx, 24*time.Hour, 3*time.Hour),
+		pipedEnvsCache:            memorycache.NewTTLCache(ctx, 24*time.Hour, 3*time.Hour),
```

Comment:
Btw, caching this piped's Envs by this way causes data inconsistency when the user updated the env list of a piped.

Reply:
So to be simple, we just validate whether the requested Env belongs to the piped's project or not.

Reply:
That seems to be good